pkg/metrics/producers/scheduledcapacity/crontabs.go

```diff
@@ -0,0 +1,77 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduledcapacity
+
+import (
+	"fmt"
+	"github.com/awslabs/karpenter/pkg/apis/autoscaling/v1alpha1"
+	"github.com/robfig/cron/v3"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Crontab string
+
+func (tab *Crontab) String() string {
+	return string(*tab)
+}
+
+// crontabFrom returns a crontab equivalent to the strongly-typed schedule
+func crontabFrom(pattern *v1alpha1.Pattern) *Crontab {
+	c := Crontab(fmt.Sprintf("%s %s %s %s %s", strings.Join(crontabFieldFrom(pattern.Minutes, true), ","),
+		strings.Join(crontabFieldFrom(pattern.Hours, true), ","), strings.Join(crontabFieldFrom(pattern.Days, false), ","),
+		strings.Join(crontabFieldFrom(pattern.Months, false), ","), strings.Join(crontabFieldFrom(pattern.Weekdays, false), ",")))
+	return &c
+}
+
+// crontabFieldFrom returns a field of a strongly-typed format into a crontab field
+func crontabFieldFrom(field *string, isZeroDefault bool) []string {
```

Comment:
Consider a default string instead of bool. That way the caller can declare its default and this function doesn't need to know.

Reply:
got it. I can't imagine that this will be used anywhere else, but I can see the benefit.